contracts/src/rollup/RollupUserLogic.sol

```diff
@@ -219,26 +219,37 @@ contract RollupUserLogic is RollupCore, UUPSNotUpgradeable, IRollupUser {
     /**
      * @notice Refund a staker that is currently staked on an assertion that either has a chlid assertion or is the latest confirmed assertion.
      */
-    function returnOldDeposit() external override onlyValidator whenNotPaused {
+    function returnOldDeposit() external override onlyValidator(msg.sender) whenNotPaused {
         requireInactiveStaker(msg.sender);
         withdrawStaker(msg.sender);
     }
 
+    /**
+     * @notice From the staker's withdrawal address, 
+     * refund a staker that is currently staked on an assertion that either has a chlid assertion or is the latest confirmed assertion.
+     */
+    function returnOldDepositFromWithdrawalAddress(address stakerAddress) external override onlyValidator(stakerAddress) whenNotPaused {
```

Comment:
nit: rename to `returnOldDepositFor`

Reply:
c2ab21e77e37803b1a5b8b14269dd04adf9d40b3